lib/cocoapods/target/pod_target.rb

```diff
@@ -290,7 +290,7 @@ def recursive_dependent_targets
     #         The name of the build configuration.
     #
     def include_in_build_config?(target_definition, configuration_name)
-      key = [target_definition.label, configuration_name]
+      key = [target_definition, configuration_name]
```

Comment:
@segiddins another way around this is we cache the `label` function in `target_definition.rb` and we keep this the same.